netsim/modules/bgp.py

```diff
@@ -14,6 +14,32 @@
 from ..augment import devices
 from ..utils import log, routing as _rp_utils
 
+BGP_DEFAULT_SESSIONS: typing.Dict[str,list] = {
+}
+
+BGP_VALID_SESSION_TYPE: list = []
+
+BGP_DEFAULT_COMMUNITY_KW: typing.Final[dict] = {
+  'standard': 'standard',
+  'extended': 'extended'
+}
+
+BGP_INHERIT_COMMUNITY: Box
+
+def setup_bgp_constants(topology: Box) -> None:
+  global BGP_DEFAULT_SESSIONS, BGP_VALID_SESSION_TYPE, BGP_INHERIT_COMMUNITY
+
+  # Valid session types are reconstructed from the valid values of the bgp.sessions.ipv4 global attribute
+  # We're assuming there's no difference between IPv4 and IPv6
+  BGP_VALID_SESSION_TYPE = [ v for v in topology.defaults.bgp.attributes['global'].sessions.ipv4 ]
```

Comment:
I think we should consider adding a method to api/__init__.py for a plugin to add a session type. It will need to be added to both ```ipv4``` and ```ipv6``` for both ```sessions``` and ```activate```

We could import a method from the bgp module to keep the logic where it belongs, but I'd rather invoke a method that lives in ```bgp.py``` (i.e. this file) rather than hard-code the structure inside the plugin

Reply:
Create a custom type to validate BGP sessions, so the change would have to be made in a single place.